File: education_app/lib/features/upload/data/models/entity_search_result.dart
class EntitySearchResult {
  final int id;
  final String title;
  final String? existingImageUrl;

  EntitySearchResult({
    required this.id,
    required this.title,
    this.existingImageUrl,
  });

  factory EntitySearchResult.fromJson(Map<String, dynamic> json) {
    return EntitySearchResult(
      id: json['id'],
      title: json['title'],
      existingImageUrl: json['existingImageUrl'],
    );
  }
}

File: Backend/DigitalStore.Domain/Entities/EntitySearchResult.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace DigitalStore.Domain.Entities
{
    [NotMapped]
    public class EntitySearchResult
    {
        public int Id { get; set; }
        public required string Title { get; set; }
        public string? ExistingImageUrl { get; set; }
    }
}
